src/main/python/janusgraph_python/core/datatypes/Circle.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2018 JanusGraph Python Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class Circle(object):
+    def __init__(self, longitude, latitude, radiusInKM):
+        """
+            The actual class representing Geographical Circle.
+
+        Args:
+            latitude (float): The latitude of circle's center.
+            longitude (float): The longitude of circle's center
+            radiusInKM (int): The radius of circle in KM.
+        """
+
+        self.latitude = float(latitude) * 1.0
+        self.longitude = float(longitude) * 1.0
+        self.radius = radiusInKM
+
+        status = self.__are_valid_coordinates()
+
+        if not status:
+            raise ValueError("Invalid Coordinates/Radius passed. "
+                             "Latitude needs to be b/w [-90, 90] and Longitude b/w [-180, 180] and Radius > 0")
+
+        pass
+
+    def getShape(self):
+        """
+            Return the Shape of Geometrical shape
+        Returns:
+            str
+        """
+        return "CIRCLE"
+
+    def toString(self):
+        """
+            Returns the String representation of Circle object.
+        Returns:
+            str
+        """
+        return "CIRCLE(lat: {}, lon: {}, r: {})".format(self.getLatitude(), self.getLongitude(), self.getRadius())
+
+    def getLatitude(self):
```

Comment:
Aren't getters like these extremely unusual as the attributes can be accessed directly?

Reply:
Yes, but what would be your suggestion? Make them privates objects of class? Bit of nitpick cleaning is to be done, hence thanks for reviewing those set of classes to point me there :-)

Reply:
Isn't the pythonic way to simply let users access the attributes directly like `circle.latitude` instead of doing `circle.getLatitude()`?

Reply:
@FlorianHockmann : Well, they can be made accessible, but why should we? When a user creates a Circle object, he is essentially creating a JanusGraph GeoShape. The latitude, longitude etc are its intrinsic properties, and I feel shouldn't be made public direction. Hence I added the methods for all 3 class attributes. Rather I made the latitude, longitude, radius are private variables. Let me know if that seems a good reason to keep it this way. :-)

Reply:
Well, I see that the reason Element class doesn't have any getttrs for its child class is because the objects are being accessed from their child. The `id` and `label` are accessed from child object. If we plan to use same structure, I was thinking, maybe something like this?
Parent class would be class named GeoShape with few methods like coordinate validity, etc. You will have multiple child classes corresponding to each GeoShape like Point and Circle, Box etc which overrides methods if needed. What do you say?

But certainly, if we can get opinions from any Python developer more experienced than me (I'm pretty new) it will be great :-)

Reply:
I'm not sure what you mean. `id` and `label` are not only used by child classes of `Element`, but also by users of these classes.

A base class could make sense in general (irrespective of of how the variables are implemented) as the Geo predicates seem to take basically all Geoshapes and not just some of them. [The doc comment for `Geo.within()`](https://github.com/JanusGraph/janusgraph/blob/54a938d3679633128637b7362cb5046e82cd5b9c/janusgraph-core/src/main/java/org/janusgraph/core/attribute/Geo.java#L86) for example:

> Whether one geographic region is completely within another

indicates that arbitrary Geoshapes can be used. I guess that can best be handled if we have a common base class (or maybe an interface in other languages) for the different Geoshapes.

Reply:
Got that. Looks like yes the users of `Element` class can access the attribute directly too. Like I do for accessing Vertex ID or Edge ID.

Ah. Looks like we are on same page regarding having a base class, and multiple GeoShapes inheriting from that class, corresponding to Point, Circle etc. If we do agree on the same, should I create an issue to track the same? I can close out the issue as part of our next minor release, where I'm planning to add schema element creation to library (Which is 70-80% done)

Reply:
I can still do that, but then I'll have to update the unit tests accodingly, and bit of code too. If we seriously want not to have methods just to retrive values, and rather use getters, then I can create an issue to track the same, as that is bit more extensive work.